lib/src/element/dial.cpp

```diff
@@ -66,7 +66,27 @@ namespace cycfi { namespace elements
    {
       if (track_info.current != track_info.previous)
       {
-         double new_value = value_from_point(ctx, track_info.current);
+         auto const& theme = get_theme();
+         double new_value;
+
+         if (theme.dial_mode == dial_mode::radial)
+         {
+            new_value = value_from_point(ctx, track_info.current);
```

Comment:
For consistency, I suggest renaming this `radial_value_from_point` and putting the code in the else branch in a `linear_value_from_point function`.

Reply:
This implementation does not have a direct relation point->value in the linear case.
It's the accumulation of the offsets between one point and the next, with a multiplier applied.
When this takes into account the click origin and the shift modifier, the value will be varied in a stateful manner.

Reply:
Ah, of course. Let's just leave this as-is.

Reply:
note: I also look at the way how vstgui does it.
While this is not the carbon copy of how vstgui processes the knob, they have also `valueFromPoint`, which returns the radial value regardless of the knob's setting.

A thing I believe they do, it's if the knob is dragged off-range, you have to turn it back by the same amount to get it back to normal range.
It's not the case in this knob, I have skipped implementing it, as it needs an additional state variable. I didn't consider the difference to be significant enough to bother.